### PR TITLE
fix: reject config values that silently produce an incomparable number

### LINKED ISSUE #233: crop_border is unvalidated: a negative value silently scores uncropped
## The gap

`SREvalConfig.__post_init__` validates every field it carries — `psnr_channels`, `ssim_channels`,
`ssim_impl`, `perceptual_metrics`, `lpips_net` — each with a message naming the YAML key to fix.

`crop_border` is validated nowhere.

## What a wrong value does

Measured on two genuinely different 32x32 images:

| `crop_border` | PSNR-RGB | behaviour |
|---:|---:|---|
| `0` | 7.852217 dB | no crop |
| `4` | 7.984293 dB | crops 4 px per side |
| `-4` | **7.852217 dB** | **silently identical to 0 — no crop, no error** |
| `-1` | **7.852217 dB** | same |
| `99` on an 8x8 image | — | raw `RuntimeError` from inside torch, naming no config key |

A negative border produces **a number that looks valid and is not comparable to any published
result.** Two figures here are comparable only if computed the same way, and this one quietly
is not.

## Why it is easy to hit

`None` is now the "derive it from the model's scale" sentinel. So the field's live values are:
`None` (auto), positive (crop), `0` (no crop) — and `-1`, the value someone reaches for when they
mean "auto" and have not read the docstring, sits silently in the same space.

The oversized case is the milder half but the same shape: a config error surfacing as a torch
tensor-dimension `RuntimeError` from several frames down, rather than as the actionable message
every sibling field gets.

## Proposal

Validate in `__post_init__`, matching the existing messages' style. Two rules:

- reject negative values outright — there is no reading under which one is meaningful
- the oversized case cannot be caught at config time (no image is in scope yet), so it belongs at
  scoring time where the tensor is: raise naming `crop_border`, the border, and the image size

## Acceptance

- [ ] A negative `crop_border` raises at construction, naming the YAML key
- [ ] `None`, `0` and positive values are unaffected
- [ ] A border that would consume the whole image raises at scoring time, naming the value and the
      image size, instead of a bare torch `RuntimeError`
- [ ] Tests for all three, proven RED first

### PR DESCRIPTION
**Pull request by YeapJieShen**
**Stack position 1 of 12 · review group: metric-path P2**

Closes #233, #238, #243.

### Why these three are one PR

They are one chain, not three fixes. `crop_border` accepting a negative value is the
defect; `scale` accepting a negative value is *how it gets there*; and #243 is the same
shape one class over — a validation that exists and does not run.

```
training_config.scale = -3         (unvalidated)
  -> SRCNN declares no `scale` hparam, so validate_against compares nothing
  -> SRLightning resolves eval_config.crop_border = -3
  -> SRScorer.crop: `if n <= 0: return tensors`
  -> every PSNR/SSIM scored on the FULL image, no error, no warning
```

That last number looks valid and is comparable to nothing.

### What changed

| field | before | now |
|---|---|---|
| `eval_config.crop_border` | `-4` scored uncropped, identical to `0` | raises, naming the YAML key |
| `training_config.scale` | `-3` / `0` accepted | must be a positive int when set |
| `layer_lrs` | length checked at optimizer build, values never | every entry, and both halves of a pair, must be positive |
| `init_std` | `-0.001` reached `torch.nn.init.normal_` | must be positive (`init_mean` stays free) |
| `init_strategy` | `'Paper'` silently selected PyTorch's default init | must be `'default'` or `'paper'` |
| `adversarial_weight` | `-1e-3` inverted the generator's objective | must be `>= 0` (zero is the content-only ablation) |
| `SRGANTrainingConfig.__post_init__` | replaced its parent's; the `compile_mode` guard never ran | calls `super()` first |

### Two guards, deliberately

`SRLightning` **assigns** `eval_config.crop_border` when it resolves the derive-from-scale
sentinel, which bypasses `__post_init__` entirely. So config validation cannot be the only
guard — `SRScorer.crop` refuses a negative border as well, and the oversized case (which
config time cannot see, because no image is in scope there) raises where the tensor is,
naming both the border and the image size instead of a bare torch `RuntimeError`.

### Evidence

- **9 tests, each proven RED on this branch's parent** before the fix.
- `683 passed` across `tests/training`, `tests/metrics`, `tests/models`.
- **All 34 config objects** across the shipped templates and the run-config mirror still
  construct unchanged — no shipped config changes behaviour.

### On #243

Not from the review passes — found while reading `__post_init__` for this PR. Measured:
`SRGANTrainingConfig(compile_mode='max-autotune', compile_backend=None)` did not raise,
while the identical construction on the parent did. An audit of every `__post_init__` in
the package found this the **only** override skipping its parent.

### CODE REVIEW

**Review by YeapJieShen — COMMENTED** [review]
Human reviewed, all good.
